src/octoprint/util/comm.py

```diff
@@ -2825,11 +2858,13 @@ def _get_next_from_job(self):
 			if isinstance(self._currentFile, StreamingGcodeFileInformation):
 				self._finishFileTransfer()
 			else:
-				self._changeState(self.STATE_FINISHING)
-				self.sendCommand("M400", part_of_job=True)
-				self._callback.on_comm_print_job_done()
 				def finalize():
-					self._changeState(self.STATE_OPERATIONAL)
+					self._changeState(self.STATE_FINISHING)
```

Comment:
Why the double decoupling here? And the change in the first place? Feels unrelated to the PR.

Reply:
because of asynchronous transmission, at the moment of read end-of-file there will be a lot of commands in queue, actual printing state must be synchronized with printing queue

Reply:
Please document that then.